MinnPost/object-sync-for-salesforce
# 276 choose date format

## What does this PR do?

This does the date format part only. Specifically this closes #276 if it's not already closed. It's going into the fieldmap design changes, #328, before going into master. 

## How do I test this PR?

- On a fieldmap, pick a Salesforce date field. Enter a valid PHP date format.
- Make sure that date format is used when data is saved in WordPress.
- Change it to a non date field. Make sure nothing weird happens.
